datalad/support/annexrepo.py

```diff
@@ -3435,18 +3436,31 @@ def pipe_data_received(self, fd, data):
             # let the base class decide what to do with it
             super().pipe_data_received(fd, data)
             return
+        if self._remainder:
+            data = self._remainder + data
+            self._remainder = None
         # this is where the JSON records come in
         # json_loads() is already logging any error, which is OK, because
         # under no circumstances we would expect broken JSON
-        for line in data.splitlines():
+        lines = data.splitlines()
+        for iline, line in enumerate(lines):
             try:
                 j = json_loads(line)
-            except Exception:
-                # TODO turn this into an error result, or put the exception
-                # onto the result future -- needs more thought
+            except Exception as exc:
                 if line.strip():
                     # do not complain on empty lines
-                    lgr.error('Received undecodable JSON output: %s', data)
+                    if iline == len(lines) - 1 and not data.endswith(os.linesep.encode()):
+                        lgr.debug("Caught %s while trying to parse JSON line %s which might "
+                                  "be not yet a full line", exc, line)
+                        # it is the last line and fails to parse -- it can/likely
+                        # to happen that it was not a complete line and that buffer
+                        # got filled up/provided before the end of line.
+                        # So - memoize it and pass to the next call
```

Comment:
Perhaps not something that will trip up other readers, but I found "memoize" and "pass" in this context confusing.  Maybe "store it so that it can be prepended to `data` in the next call"?